Index: app/src/main/java/com/example/happybirthday/MainActivity.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/com/example/happybirthday/MainActivity.kt b/app/src/main/java/com/example/happybirthday/MainActivity.kt
--- a/app/src/main/java/com/example/happybirthday/MainActivity.kt	
+++ b/app/src/main/java/com/example/happybirthday/MainActivity.kt	(date 1676071333592)
@@ -3,7 +3,6 @@
 import android.os.Bundle
 import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
-import androidx.compose.foundation.layout.Column
 import androidx.compose.material.MaterialTheme
 import androidx.compose.material.Surface
 import androidx.compose.material.Text
@@ -12,10 +11,16 @@
 import androidx.compose.ui.unit.sp
 import com.example.happybirthday.ui.theme.HappyBirthdayTheme
 import androidx.compose.foundation.Image
-import androidx.compose.foundation.layout.Box
-import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.*
+import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.layout.ContentScale.Companion.Crop
 import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
 import com.example.happybirthday.R
 
 class MainActivity : ComponentActivity() {
@@ -25,7 +30,15 @@
             HappyBirthdayTheme {
                 // A surface container that uses the 'background' color from the theme
                 Surface(modifier = Modifier.fillMaxSize(), color = MaterialTheme.colors.background) {
-                    BirthdayGreetingWithImage( "Happy Birthday Sam!", "- from Emma")
+                    BirthdayGreetingWithImage(
+
+                        // save your texts into a stringResource
+                        // and you can use it anywhere
+                        // you only need to update it in the R
+                        stringResource(R.string.happy_birthday_text),
+                        stringResource(R.string.signature_text)
+
+                    )
                 }
             }
         }
@@ -34,32 +47,75 @@
 
 @Composable
 fun BirthdayGreetingWithText(message: String, from: String) {
-    Column {
+    Column (
+
+        // Modifier.fillMaxSize() will set the height and width of the column layout
+        // to the maximum available height and width
+        // in other words to full screen
+        modifier = Modifier.fillMaxSize(),
+        // try to run this code with/without the above line
+
+        // put the content (two texts) of this column to the bottom
+        verticalArrangement = Arrangement.Bottom,
+
+        // put the content of this column to the center
+        horizontalAlignment = Alignment.CenterHorizontally)
+
+
+    {
         Text(
             text = message,
             fontSize = 36.sp,
+
+            // add a green background only to this text box
+            modifier = Modifier
+                .background(color = Color.Green)
+                .align(alignment = Alignment.End)
+
+
         )
         Text(
             text = from,
             fontSize = 24.sp,
+
+            // add a 50 dp size padding to the top of this text
+            modifier = Modifier
+                .background(color = Color.Red)
+                .padding(top = 70.dp)
+
         )
     }
 }
 
 @Composable
-fun BirthdayGreetingWithImage(message: String, from: String){
+fun BirthdayGreetingWithImage(message: String, from: String, modifier: Modifier = Modifier){
+
+    // Use the painterResource to load picture formats like PNGs.
     val image = painterResource(id = R.drawable.androidparty)
+
     Box {
-        Image(painter = image, contentDescription = null)
+        Image(painter = image,
+            contentDescription = null,
+
+            // see the effect with and without the below line
+            contentScale = Crop
+
+        )
         BirthdayGreetingWithText(message = message, from = from)
     }
 }
 
 
-@Preview(showBackground = true)
+@Preview(showBackground = false)
 @Composable
 fun BirthdayCardPreview() {
     HappyBirthdayTheme {
-        BirthdayGreetingWithImage( "Happy Birthday Sam!", "- from Emma")
+        BirthdayGreetingWithImage(
+
+            // use the stringResource
+            stringResource(R.string.happy_birthday_text),
+            stringResource(R.string.signature_text)
+
+        )
     }
 }
\ No newline at end of file
